SampleScripts/ppm/passwordpolicymanager.ps1

```diff
@@ -0,0 +1,1456 @@
+	Param (
+		[Parameter (Mandatory = $true)] [ValidateNotNullOrEmpty()] [String]$server,
+        [Parameter (Mandatory = $true)] [ValidateNotNullOrEmpty()] [String]$user,
+        [Parameter (Mandatory = $true)] [ValidateNotNullOrEmpty()] [String]$pass,
+		[Parameter (Mandatory = $true)] [ValidateNotNullOrEmpty()] [String]$domain,
+		[Parameter (Mandatory = $true)] [ValidateNotNullOrEmpty()] [String]$cluster
+	)
+	
+	#construct Environment Object
+	$passwordPolicyNodes = New-Object System.Collections.Generic.List[System.Object]
+	$results = Get-EsxiPasswordPolicy -server $server -user $user -pass $pass -domain $domain -cluster $cluster
+	foreach ($result in $results) {
+		$esxiConfigTypeValues =  New-Object -TypeName psobject
+		$esxiConfigTypeValues | Add-Member -notepropertyname "passwdExpInDays" -notepropertyvalue "Empty"
+		$esxiConfigTypeValues | Add-Member -notepropertyname "passwdMaxFailAttempts" -notepropertyvalue "Empty"
+		$esxiConfigTypeValues | Add-Member -notepropertyname "passwdMinimumLengthFor1CharClass" -notepropertyvalue "Empty"
+		$esxiConfigTypeValues | Add-Member -notepropertyname "passwdMinimumLengthFor2CharClass" -notepropertyvalue "Empty"
+		$esxiConfigTypeValues | Add-Member -notepropertyname "passwdMinimumLengthFor3CharClass" -notepropertyvalue "Empty"
+		$esxiConfigTypeValues | Add-Member -notepropertyname "passwdMinimumLengthFor4CharClass" -notepropertyvalue "Empty"
+		$esxiConfigTypeValues | Add-Member -notepropertyname "passwdMinimumCharLengthForPhrase" -notepropertyvalue "Empty"
+
+		#parsing ESXi password policy string
+		$result.PasswordQualityControl | Select-String -Pattern "^retry=(\d+)\s+min=(.+),(.+),(.+),(.+),(.+)" | Foreach-Object {$PasswdPolicyRetryValue, $PasswdPolicyMinValue1, $PasswdPolicyMinValue2, $PasswdPolicyMinValue3, $PasswdPolicyMinValue4, $PasswdPolicyMinValue5 = $_.Matches[0].Groups[1..6].Value}
+		$esxiConfigTypeValues.passwdExpInDays = $result.PaswordMaxDays
+		$esxiConfigTypeValues.passwdMaxFailAttempts = $PasswdPolicyRetryValue
+		$esxiConfigTypeValues.passwdMinimumLengthFor1CharClass = $PasswdPolicyMinValue1
+		$esxiConfigTypeValues.passwdMinimumLengthFor2CharClass = $PasswdPolicyMinValue2
+		$esxiConfigTypeValues.passwdMinimumLengthFor3CharClass = $PasswdPolicyMinValue4
+		$esxiConfigTypeValues.passwdMinimumLengthFor4CharClass = $PasswdPolicyMinValue5
+		$esxiConfigTypeValues.passwdMinimumCharLengthForPhrase = $PasswdPolicyMinValue3
+
+		$nodeConfigValue = New-Object -TypeName psobject
+		$nodeConfigValue | Add-Member -notepropertyname "FQDN" -notepropertyvalue $result.fqdn
+		$nodeConfigValue | Add-Member -notepropertyname "productType" -notepropertyvalue "ESXi"
+		$nodeConfigValue | Add-Member -notepropertyname "productAttributes" -notepropertyvalue $esxiConfigTypeValues
+		$passwordPolicyNodes.Add($nodeConfigValue)
+		Remove-Variable -Name nodeConfigValue
+		Remove-Variable -Name esxiConfigTypeValues
+	}
+	return $passwordPolicyNodes
+}
+
+Function GetVCServerPasswordPolicy {
+	Param (
```

Comment:
Formatting, all Parameters in line

Reply:
addressed in 9ee41934dd0ee5a302dd128339b72e2f82898175